### PR TITLE
[VEN-1380] Optimised swap router to reduce the size under 24 KB

### PR DESCRIPTION
**Pull request by defcon022**
## Description

Resolves #

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/Swap/SwapRouter.sol [hunk 1]
```diff
@@ -717,4 +695,31 @@ contract SwapRouter is Ownable2Step, RouterHelper, IPancakeSwapV2Router {
             revert RepayError(msg.sender, vTokenAddress, response);
         }
     }
+
+    /**
+     * @notice check for min amount out.
+     * @param path The addresses of the underlying token
+     * @param balanceBefore Balance before the swap.
+     * @param amountOutMin Min amount out threshold.
```
Comment (chechu) [inline contracts/Swap/SwapRouter.sol:703]: I miss the `to` param doc. The `@notice` fragment could be: check if the balance of `to` minus the balanceBefore is greater than the amountOutMin

File: contracts/Swap/SwapRouter.sol [hunk 2]
```diff
@@ -717,4 +695,31 @@ contract SwapRouter is Ownable2Step, RouterHelper, IPancakeSwapV2Router {
...
+        uint256 amountOutMin,
+        address to
+    ) internal view returns (uint256 swapAmount) {
+        uint256 balanceAfter = IERC20(path[path.length - 1]).balanceOf(to);
+        swapAmount = balanceAfter - balanceBefore;
+        require(swapAmount >= amountOutMin, "SwapRouter: SwapAmount is less than amountOutMin");
+    }
+
+    /**
+     * @notice Get swapped amount
```
Comment (chechu) [inline contracts/Swap/SwapRouter.sol:717]: We should try to write better docs. Here something like "returns the difference between the balance of `this` and the balanceBefore" could be better

File: contracts/Swap/SwapRouter.sol (hunk 1 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol:703]: Fixed, 6ef9faf8fc19a066c7e4a7dac440a9bd25f2a1cc

File: contracts/Swap/SwapRouter.sol (hunk 2 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol]: Fixed, 6ef9faf8fc19a066c7e4a7dac440a9bd25f2a1cc

**Review by chechu — APPROVED** [review]

**Pull request merged**